src/olympia/users/models.py

```diff
@@ -669,16 +669,21 @@ def allow_request(cls, request):
         """
         try:
             remote_addr = ipaddress.ip_address(request.META.get('REMOTE_ADDR'))
+            if (request.user):
+                user_last_login_ip = ipaddress.ip_address(
+                    request.user.last_login_ip)
         except ValueError:
             # If we don't have a valid ip address, let's deny
             return False
 
         restrictions = IPNetworkUserRestriction.objects.all()
 
         for restriction in restrictions:
-            if remote_addr in restriction.network:
-                log.info('Restricting request from %s %s (%s)',
+            if (remote_addr in restriction.network or
+               user_last_login_ip in restriction.network):
+                log.info('Restricting request from %s %s, %s %s (%s)',
                          'ip', remote_addr,
+                         'user last login ip', user_last_login_ip,
```

Comment:
Maybe naming this `last_login_ip` instead of the spaces makes it easier to match later on?